include/nbl/asset/ICPUDescriptorSet.h

```diff
@@ -108,30 +122,43 @@ class NBL_API ICPUDescriptorSet final : public IDescriptorSet<ICPUDescriptorSetL
 			{
                 --referenceLevelsBelowToConvert;
 				m_layout->convertToDummyObject(referenceLevelsBelowToConvert);
-				for (auto it=m_descriptors->begin(); it!=m_descriptors->end(); it++)
+
+				for (uint32_t t = 0u; t < EDT_COUNT; ++t)
 				{
-					auto descriptor = it->desc.get();
-					if (!descriptor)
+					const auto type = static_cast<E_DESCRIPTOR_TYPE>(t);
+					const auto descriptorCount = m_layout->getTotalDescriptorCount(type);
+					if (descriptorCount == 0ull)
 						continue;
-					switch (descriptor->getTypeCategory())
+
+					auto descriptors = m_descriptors[type]->begin();
+					assert(descriptors);
+
+					for (uint32_t i = 0u; i < descriptorCount; ++i)
 					{
+						switch (descriptors[i]->getTypeCategory())
+						{
 						case IDescriptor::EC_BUFFER:
-							static_cast<asset::ICPUBuffer*>(descriptor)->convertToDummyObject(referenceLevelsBelowToConvert);
+							static_cast<asset::ICPUBuffer*>(descriptors[i].get())->convertToDummyObject(referenceLevelsBelowToConvert);
 							break;
+
 						case IDescriptor::EC_IMAGE:
-							static_cast<asset::ICPUImageView*>(descriptor)->convertToDummyObject(referenceLevelsBelowToConvert);
-							if (descriptor->getTypeCategory()==IDescriptor::EC_IMAGE && it->image.sampler)
-								it->image.sampler->convertToDummyObject(referenceLevelsBelowToConvert);
-							break;
+						{
+							static_cast<asset::ICPUImageView*>(descriptors[i].get())->convertToDummyObject(referenceLevelsBelowToConvert);
+							const auto mutableSamplerCount = m_layout->getTotalMutableSamplerCount();
+							for (uint32_t s = 0u; s < mutableSamplerCount; ++s)
+								m_mutableSamplers->begin()[s]->convertToDummyObject(referenceLevelsBelowToConvert);
```

Comment:
pretty sure this is wrong and you need to do it a separate loop

Reply:
hmm seems that `m_mutableSamplers` also duplicate with `m_descriptorInfos`

Reply:
After merging `IEmulatedDescriptorSet::m_descriptors` and `ICPUDescriptorSet::m_descriptorInfos`, this can once again be done inline, no separate loop needed.

Reply:
Resolved.